### ISSUE TITLE
Panic when accessing HTTPS endpoint with HTTP

### ISSUE THREAD

**Issue by glebpom** [poem-web/poem#141]
## Expected Behavior
No panic if accessing HTTPS server with HTTP request

## Actual Behavior
```
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread 'reactor' panicked at 'unexpected polling after handshake', /Users/glebpomykalov/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-rustls-0.22.0/src/common/handshake.rs:38:17
```

## Steps to Reproduce the Problem

1. create rustls TLS server
2. try to request with HTTP

## Specifications

- Version: 1.2.15
- Platform: macox
- Subsystem:

**Comment by wileybaba** [comment]
this is related to [this commit](https://github.com/poem-web/poem/commit/ff56f9f5eb83c797516745f3e729c5b506da25c4)

If you revert to 1.2.12 you should be okay until this is resolved.  Looking into a fix for it now

**Comment by sunli829** [comment]
Fixed in `1.2.16`
